src/scripts/farming/Plot.ts

```diff
@@ -373,6 +373,15 @@ class Plot implements Saveable {
                 this.notifications.push(FarmNotificationType.Dropped);
             }
 
+            // Check for Banetteite drop if Kasib died
+            if (App.game.party.alreadyCaughtPokemonByName('Banette') && !player.hasMegaStone(GameConstants.MegaStoneType.Banettite)) {
+                if (this.berry == BerryType.Kasib) {
+                    if (Rand.chance(0.05)) {
+                        player.gainMegaStone(GameConstants.MegaStoneType.Banettite);
+                    }
+                }
+            }
```

Comment:
The berry should be checked first here
```suggestion
            // Check for Banetteite drop if Kasib died
            if (this.berry === BerryType.Kasib) {
                if (!player.hasMegaStone(GameConstants.MegaStoneType.Banettite) && App.game.party.alreadyCaughtPokemonByName('Banette')) {
                    if (Rand.chance(0.05)) {
                        player.gainMegaStone(GameConstants.MegaStoneType.Banettite);
                    }
                }
            }
```